src/main/java/seedu/cakecollate/model/order/ContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,119 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import seedu.cakecollate.commons.util.StringUtil;
+import seedu.cakecollate.logic.parser.Prefix;
+import seedu.cakecollate.model.tag.Tag;
+
+/**
+ * Tests that an {@code Order}'s {@code Name} matches any of the keywords given.
+ */
+public class ContainsKeywordsPredicate implements Predicate<Order> {
+    private final HashMap<Prefix, List<String>> keywords;
+
+    public ContainsKeywordsPredicate(HashMap<Prefix, List<String>> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Order order) {
+        boolean result = true;
+        for (Map.Entry<Prefix, List<String>> entry : keywords.entrySet()) {
+            Prefix prefix = entry.getKey();
+            List<String> value = entry.getValue();
+            String testString = getTestString(prefix, order);
+            result = result && prefixTest(value, testString);
+        }
+        return result;
+    }
+
+    /**
+     * Test each {@code String} retrieved from {@code Order} to see if {@code String}
+     * contains any of the provided keywords.
+     */
+    public boolean prefixTest(List<String> toTest, String toTestFrom) {
+        return toTest.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(toTestFrom, keyword));
+    }
+
+    /**
+     * For the specified {@code Order}, retrieve, process and return the {@code String} to test keywords against
+     * based on the provided {@code Prefix}.
+     */
+    public String getTestString(Prefix prefix, Order order) {
+        // Cannot use switch-case unless I edit the Prefix class...
+        if (prefix.equals(PREFIX_NAME)) {
+            return order.getName().fullName;
+        } else if (prefix.equals(PREFIX_PHONE)) {
+            return order.getPhone().value;
+        } else if (prefix.equals(PREFIX_EMAIL)) {
+            return order.getEmail().value;
+        } else if (prefix.equals(PREFIX_ADDRESS)) {
+            return order.getAddress().value;
+        } else if (prefix.equals(PREFIX_ORDER_DESCRIPTION)) {
+            Set<OrderDescription> orderDescriptions = order.getOrderDescriptions();
+            Set<String> orderDescriptionsString = orderDescriptions.stream()
+                    .map(OrderDescription::toString)
+                    .collect(Collectors.toSet());
+            return String.join(" ", orderDescriptionsString);
+        } else if (prefix.equals(PREFIX_TAG)) {
+            Set<Tag> tags = order.getTags();
+            Set<String> tagsString = tags.stream()
+                    .map(tag -> tag.tagName)
+                    .collect(Collectors.toSet());
+            return String.join(" ", tagsString);
+        } else if (prefix.equals(PREFIX_DATE)) {
+            LocalDate deliveryDate = order.getDeliveryDate().value;
```

Comment:
I don't really understand why a delivery date has to be returned in all formats here? 😅 Also, could any of it be abstracted away in your delivery date class?

Reply:
Uhm because I wanted to let the user be able to search for a date using any of the format they use when they add the order + if they input the day/month/year only. Though now that I look at it I probably can do away with day/month/year since they are already included in the 4 formats provided haha. Also, yeah probably can create a method in deliverydate to get formats.